### PR TITLE
[IA-2455] Update/Add Metrics for Proxy Request Counts

### PR DESCRIPTION
**Pull request by ehanna4**
Adds/updates the counting of proxy requests per the new naming convention.

Name: proxyRequest
- result
- action
- tool

Name: proxyAuthRequest
- type

Also adds a metric for tracking notebook sizes on notebook saves
Name: notebooksSize
- source

Tested through a fiab - please see https://console.cloud.google.com/monitoring/dashboards/builder/7c01affa-ea94-40cd-aa73-33c9e391d6f5?authuser=3&project=broad-dsp-stackdriver&dashboardBuilderState=%257B%2522editModeEnabled%2522:true%257D&timeDomain=1h for WIP graphs showing the metrics coming through

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: I think this method is only called from the `"jupyter" / "terminals"` route so this `if` is redundant.. but it doesn't hurt.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: I think we may have an ADT for this .. `RuntimeContainerServiceType`. 

I wonder if this would work (I haven't compiled it :)):
```
tool = RuntimeContainerServiceType.values.find(s => request.uri.toString.contains(s.proxySegment)).map(_.imageType.name).getOrElse("other")
```

I don't feel too strongly, but seems nice to not duplicate the proxy segment / tool name strings.

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: I think `.getContentLengthOption().getAsLong` is the akka Java API, and probably throws an exception if the optional is not present.

I would do:
```
resp.entity.contentLengthOption.traverse { c =>
  metrics.gauge("proxy/notebooksSize", c.toDouble)
}
```
which is the Scala API and safer `Option`-handling. (I don't think it's too important to check the `.toDouble` conversion.)

Comment (rtitle) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: Also just realized this is using the `HttpResponse`. Do we want to do it on the `HttpRequest` instead? (Since I think the PUT _request_ would have the right content-length.)

Comment (ehanna4) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: oops thank you yes should be the request

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (ehanna4) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: oh yes that's much better thanks

**Review by rtitle — APPROVED** [review]
👍

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (gpcarr) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: going forward we want to keep metric names to a single string (with no additional `/`s) after `OpenCensus/leonardo/` so that we have a uniform naming pattern. We can think about extracting as much as possible into tags so maybe for this metric name, we can use `notebookSize` with tag `source: proxy` or something like that. then going forward we can add other notebook attribute metrics with same or different sources

**Review by gpcarr — APPROVED** [review]
LGTM 👍 would be curious to know if you're able to create the graphs mentioned in the original ticket under project `broad-dsde-dev` (using your fiab) with the tags added in this PR. could be a good thing to test before merging

**Comment by ehanna4** [comment]
> LGTM 👍 would be curious to know if you're able to create the graphs mentioned in the original ticket under project `broad-dsde-dev` (using your fiab) with the tags added in this PR. could be a good thing to test before merging

Do you have access with this link: https://console.cloud.google.com/monitoring/dashboards/builder/7c01affa-ea94-40cd-aa73-33c9e391d6f5?authuser=3&project=broad-dsp-stackdriver&dashboardBuilderState=%257B%2522editModeEnabled%2522:true%257D&timeDomain=1h

I've got some WIP graphs made for dev/fiab

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala
Comment (Qi77Qi) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/ProxyRoutes.scala]: maybe use `api` instead of `type`? `type` sounds super vague to me

**Pull request merged**
